src/software/new_geom/convex_polygon.h

```diff
@@ -10,13 +10,30 @@ class ConvexPolygon : public Polygon, public ConvexShape
 {
    public:
     ConvexPolygon() = delete;
+
     /**
      * Returns the area of this convex polygon
      *
      * @return The area of this convex polygon
      */
     double area() const override;
 
+    /**
+     * Returns the centroid of this convex polygon
+     *
+     * @return The centroid of this convex polygon
+     */
+    Point centroid() const;
+
+    /**
+     * Returns the ConvexPolygon expanded in the direction of v
+     * The ConvexPolygon is split in half perpendicular to v and points on the half that
```

Comment:
Not super familiar with the polygon class, but would it be more accurate to say 'split along the centroid of the shape' than 'in half'?

Reply:
Yeah that's right. Looking at how I implemented `expand` for `Rectangles` (first expand x component, then y component), I think I should do the same for `ConvexPolygon` because the implementations would be consistent and we won't have to reason about shapes being cut along weird axes. @EvanMorcom @garethellis0 @MathewMacDougall What do you think?

Reply:
The rectangle is still split before it's expanded, it's just easier to reason about since it's a more well-defined shape.

@jonathanlew are you just suggesting applying the "expansion" to each point in the same way as rectangle (ie. by x/y component individually) without any change to _what_ points are expanded?

Reply:
@MathewMacDougall There is a difference between the two types of expansion. Really wish I could whiteboard this but here's a beautiful paint drawing to explain
![image](https://user-images.githubusercontent.com/31377876/82502142-5a1a9c80-9aab-11ea-9279-9fe4ccf9fd76.png)

Reply:
I feel like the bottom case would be the more expected behaviour IMO

Reply:
Then that's a problem for Rectangle::expand. I'll change that to only do vertical and horizontal

Reply:
Oh nice drawing @jonathanlew . I didn't realize in the ConvexPolygon case we'd take the points on the "side" of the centroid for each component.

Reply:
Wait, so which method is used for ConvexPolygon and Rectangle?